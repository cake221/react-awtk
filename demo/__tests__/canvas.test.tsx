import "../canvas"

describe("test canvas", ()=>{

  it("toMatchSnapshot", ()=>{
    expect(__AwtkSnapshot).toMatchSnapshot()
  })
})
