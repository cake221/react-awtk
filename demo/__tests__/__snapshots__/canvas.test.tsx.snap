// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test canvas toMatchSnapshot 1`] = `
Array [
  "getRootHostContext",
  "getChildHostContext",
  "shouldSetTextContent",
  "createInstance",
  "funtion  window_create( parent, x, y, w, h ) :null, 0, 0, 0, 0",
  "funtion  widget_set_name( widget, name ) :undefined, vgcanvas",
  "finalizeInitialChildren",
  "prepareForCommit",
  "appendNodeChild",
  "resetAfterCommit",
  "getPublicInstance",
  "funtion  widget_child( widget, name ) :undefined, paint_vgcanvas",
  "funtion  widget_on( widget, type, on_event, ctx ) :undefined, EVT_PAINT, function (e) {
            var evt = src_1.TPaintEvent.cast(e);
            var c = src_1.TCanvas.cast(evt.c);
            var vg = c.getVgcanvas();
            c.setFillColor(\\"#f0f0f0\\");
            c.fillRect(0, 0, c.getWidth(), c.getHeight());
            vg.setFillColor(\\"green\\");
            vg.rect(10, 10, vg.w - 20, vg.h - 20);
            vg.fill();
            vg.save();
            vg.setLineWidth(3);
            vg.setStrokeColor(\\"blue\\");
            vg.rect(30, 30, vg.w - 60, vg.h - 60);
            vg.stroke();
            vg.restore();
        }, null",
]
`;
